controllers/actions.github.com/autoscalinglistener_controller.go

```diff
@@ -345,7 +380,43 @@ func (r *AutoscalingListenerReconciler) createServiceAccountForListener(ctx cont
 }
 
 func (r *AutoscalingListenerReconciler) createListenerPod(ctx context.Context, autoscalingRunnerSet *v1alpha1.AutoscalingRunnerSet, autoscalingListener *v1alpha1.AutoscalingListener, serviceAccount *corev1.ServiceAccount, secret *corev1.Secret, logger logr.Logger) (ctrl.Result, error) {
-	newPod := r.resourceBuilder.newScaleSetListenerPod(autoscalingListener, serviceAccount, secret)
+	var envs []corev1.EnvVar
+	if autoscalingListener.Spec.Proxy != nil {
+		httpURL := corev1.EnvVar{
+			Name: "http_proxy",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{Name: proxyListenerSecretName(autoscalingListener)},
+					Key:                  "http_proxy",
+				},
+			},
+		}
+		envs = append(envs, httpURL)
```

Comment:
should we add each env as needed?
ex:
```
if autoscalingListener.Spec.Proxy.HTTP != nil {
  envs = append(envs, httpURL)
}
```

Reply:
Yeah I think that would help debugging issues, right now it works as in the env var would be set to `""` if absent but it may confuse to see that being set if you didn't provide it. I'll change it for the runner too, thanks!